src/irr/asset/CGLSLGraphicsPipelineBuiltinIncludeLoader.h

```diff
@@ -0,0 +1,59 @@
+#ifndef __IRR_C_GLSL_GRAPHICS_PIPELINE_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+#define __IRR_C_GLSL_GRAPHICS_PIPELINE_BUILTIN_INCLUDE_LOADER_H_INCLUDED__
+
+#include "irr/asset/IBuiltinIncludeLoader.h"
+
+namespace irr {
+namespace asset
+{    
+
+class CGLSLGraphicsPipelineBuiltinIncludeLoader : public IBuiltinIncludeLoader
+{
+public:
+    const char* getVirtualDirectoryName() const override { return "glsl/graphicspipeline/"; }
+
+private:
+    static std::string getMTLloaderCommonDefinitions(const std::string&)
+    {
+        return
+R"(#ifndef _IRR_MTL_LOADER_COMMON_INCLUDED_
+#define _IRR_MTL_LOADER_COMMON_INCLUDED_
+
+struct irr_glsl_SMTLPushConstants 
```

Comment:
don't call PushConstants (can place in UBO or SSBO for large render systems) prefer MaterialParameters instead.